src/meta/client/MetaClient.cpp

```diff
@@ -999,27 +1008,26 @@ StatusOr<SchemaVer> MetaClient::getNewestEdgeVerFromCache(const GraphSpaceID& sp
 
 
 folly::Future<StatusOr<bool>> MetaClient::heartbeat() {
+    if (clusterId_.load() == 0) {
+        clusterId_ = ClusterIdMan::getClusterIdFromFile(FLAGS_cluster_id_path);
+    }
     cpp2::HBReq req;
     nebula::cpp2::HostAddr thriftHost;
     thriftHost.set_ip(localHost_.first);
     thriftHost.set_port(localHost_.second);
     req.set_host(std::move(thriftHost));
-    req.set_clusterId(clusterMan_->getClusterId());
-
+    req.set_cluster_id(clusterId_.load());
+    LOG(INFO) << "Send heartbeat to " << leader_ << ", clusterId " << req.get_cluster_id();
     return getResponse(std::move(req), [] (auto client, auto request) {
         return client->future_heartBeat(request);
     }, [this] (cpp2::HBResp&& resp) -> bool {
-        if (resp.code == cpp2::ErrorCode::SUCCEEDED
-            && !clusterMan_->isClusterIdSet()) {
-            clusterMan_->setClusterId(resp.get_clusterId());
-            if (!clusterMan_->clusterIdDumped()) {
-                if (!clusterMan_->dumpClusterId()) {
-                    LOG(ERROR) << "meta client clusterId dump failed!";
-                    return false;
-                }
+        if (clusterId_.load() == 0) {
+            LOG(INFO) << "Persisit the cluster Id from metad " << resp.get_cluster_id();
+            if (ClusterIdMan::persistInFile(resp.get_cluster_id(), FLAGS_cluster_id_path)) {
```

Comment:
Do we need return false when persist failed?

Reply:
Good point.